lgc/builder/DescBuilder.cpp

```diff
@@ -243,8 +511,105 @@ Value *DescBuilder::scalarizeIfUniform(Value *value, bool isNonUniform) {
 Value *DescBuilder::CreateGetBufferDescLength(Value *const bufferDesc, const Twine &instName) {
   // In future this should become a full LLVM intrinsic, but for now we patch in a late intrinsic that is cleaned up
   // in patch buffer op.
-  Instruction *const insertPos = &*GetInsertPoint();
   std::string callName = lgcName::LateBufferLength;
   addTypeMangling(nullptr, bufferDesc, callName);
-  return emitCall(callName, getInt32Ty(), bufferDesc, Attribute::ReadNone, insertPos);
+  return CreateNamedCall(callName, getInt32Ty(), bufferDesc, Attribute::ReadNone);
+}
+
+// =====================================================================================================================
+// Calculate a buffer descriptor for an inline buffer
+//
+// @param descPtr : Pointer to inline buffer
+Value *DescBuilder::buildInlineBufferDesc(Value *descPtr) {
+  // Bitcast the pointer to v2i32
+  descPtr = CreatePtrToInt(descPtr, getInt64Ty());
+  descPtr = CreateBitCast(descPtr, VectorType::get(getInt32Ty(), 2));
+
+  // Build descriptor words.
+  SqBufRsrcWord1 sqBufRsrcWord1 = {};
+  SqBufRsrcWord2 sqBufRsrcWord2 = {};
+  SqBufRsrcWord3 sqBufRsrcWord3 = {};
+
+  sqBufRsrcWord1.bits.baseAddressHi = UINT16_MAX;
+  sqBufRsrcWord2.bits.numRecords = UINT32_MAX;
+
+  sqBufRsrcWord3.bits.dstSelX = BUF_DST_SEL_X;
+  sqBufRsrcWord3.bits.dstSelY = BUF_DST_SEL_Y;
+  sqBufRsrcWord3.bits.dstSelZ = BUF_DST_SEL_Z;
+  sqBufRsrcWord3.bits.dstSelW = BUF_DST_SEL_W;
+  sqBufRsrcWord3.gfx6.numFormat = BUF_NUM_FORMAT_UINT;
+  sqBufRsrcWord3.gfx6.dataFormat = BUF_DATA_FORMAT_32;
+  assert(sqBufRsrcWord3.u32All == 0x24FAC);
+
+  // DWORD0
```

Comment:
I plan to do a change clearing some coding issues. DWORD -> Dword or dword is in the list.

Reply:
I have now changed these ones.